sdk/core/System.ClientModel/src/ModelReaderWriter/ReflectionModelInfo.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.ClientModel.Primitives;
+
+internal class ReflectionModelInfo : ModelInfo
+{
+    private Type _type;
+
+    public ReflectionModelInfo(Type type)
+    {
+        _type = type;
+    }
+
+    //This will be replaced with RequiresUnreferencedCode when this issue is fixed https://github.com/Azure/azure-sdk-for-net/issues/48294
+    [UnconditionalSuppressMessage("Trimming", "IL2077",
+      Justification = "We will only call this when we went through ModelReaderWriter.Read which has DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors).")]
+    public override object CreateObject()
```

Comment:
```suggestion
[RequiresUnreferencedCode("Uses reflection that can't be statically analyzed"]
internal class ReflectionModelInfo : ModelInfo
{
    private Type _type;

    public ReflectionModelInfo(Type type)
    {
        _type = type;
    }

    [UnconditionalSuppressMessage("Trimming", "IL2077",
      Justification = "This class is marked as RequiresUnreferencedCode.")]
    public override object CreateObject()
```

It's better to be more explicit that this whole class is not trim compatible. That way any callers are aware they need to reconsider calling this class in trimmed scenarios.

Reply:
Alternatively, if you can flow the `DynamicallyAccessedMembers` attribute to the constructor (and through `ModelReaderWriterContext.GetModelInfo`, it may work to be able to do this without a suppression.